internal/controller/pipelinerollout_controller.go

```diff
@@ -646,28 +637,45 @@ func (r *PipelineRolloutReconciler) processPipelineStatus(ctx context.Context, p
 
+	r.setChildResourcesHealthCondition(pipelineRollout, existingPipelineDef, &pipelineStatus)
+	r.setChildResourcesPauseCondition(pipelineRollout, existingPipelineDef, &pipelineStatus)
+
+	return nil
+}
+
+func (r *PipelineRolloutReconciler) setChildResourcesHealthCondition(pipelineRollout *apiv1.PipelineRollout, pipeline *kubernetes.GenericObject, pipelineStatus *kubernetes.GenericStatus) {
+
 	pipelinePhase := numaflowv1.PipelinePhase(pipelineStatus.Phase)
 	pipelineChildResourceStatus, pipelineChildResourceReason := getPipelineChildResourceHealth(pipelineStatus.Conditions)
 
-	if pipelineChildResourceReason == "Progressing" {
+	if pipelineChildResourceReason == "Progressing" || !pipelineObservedGenerationCurrent(pipeline.Generation, pipelineStatus.ObservedGeneration) {
 		pipelineRollout.Status.MarkChildResourcesUnhealthy("Progressing", "Pipeline Progressing", pipelineRollout.Generation)
-	} else if pipelinePhase == numaflowv1.PipelinePhaseFailed || pipelineChildResourceStatus == "False" {
-		pipelineRollout.Status.MarkChildResourcesUnhealthy("PipelineFailed", "Pipeline Failed", pipelineRollout.Generation)
+	} else if pipelinePhase == numaflowv1.PipelinePhaseFailed {
+		pipelineRollout.Status.MarkChildResourcesUnhealthy("PipelineFailed", "Pipeline Phase=Failed", pipelineRollout.Generation)
+	} else if pipelineChildResourceStatus == "False" {
+		pipelineRollout.Status.MarkChildResourcesUnhealthy("PipelineFailed", "Pipeline Failed, Pipeline Child Resource(s) Unhealthy", pipelineRollout.Generation)
 	} else if pipelinePhase == numaflowv1.PipelinePhasePaused || pipelinePhase == numaflowv1.PipelinePhasePausing {
-		reason := fmt.Sprintf("Pipeline%s", string(pipelinePhase))
-		msg := fmt.Sprintf("Pipeline %s", strings.ToLower(string(pipelinePhase)))
-		pipelineRollout.Status.MarkPipelinePausingOrPaused(reason, msg, pipelineRollout.Generation)
-		setPipelineHealthStatus(existingPipelineDef, pipelineRollout, pipelineStatus.ObservedGeneration)
-	} else if pipelinePhase == numaflowv1.PipelinePhaseUnknown || pipelineChildResourceStatus == "Unknown" {
-		pipelineRollout.Status.MarkChildResourcesHealthUnknown("PipelineUnknown", "Pipeline Phase Unknown", pipelineRollout.Generation)
+		pipelineRollout.Status.MarkChildResourcesHealthUnknown("PipelineUnknown", "Pipeline Pausing - health unknown", pipelineRollout.Generation)
 	} else if pipelinePhase == numaflowv1.PipelinePhaseDeleting {
 		pipelineRollout.Status.MarkChildResourcesUnhealthy("PipelineDeleting", "Pipeline Deleting", pipelineRollout.Generation)
+	} else if pipelinePhase == numaflowv1.PipelinePhaseUnknown || pipelineChildResourceStatus == "Unknown" {
+		pipelineRollout.Status.MarkChildResourcesHealthUnknown("PipelineUnknown", "Pipeline Phase Unknown", pipelineRollout.Generation)
 	} else {
-		setPipelineHealthStatus(existingPipelineDef, pipelineRollout, pipelineStatus.ObservedGeneration)
-		pipelineRollout.Status.MarkPipelineUnpaused(pipelineDef.Generation)
```

Comment:
this `pipelineDef.Generation` actually wasn't set to anything, so this was effectively not setting `observedGeneration` at all